iso15118/secc/states/iso15118_20_states.py

```diff
@@ -1769,7 +1795,39 @@ async def process_message(
             ISOV20PayloadTypes.DC_MAINSTREAM,
         )
 
-    async def build_dc_charge_loop_res(self) -> DCChargeLoopRes:
+    def update_dc_charge_loop_params(
+        self,
+        dc_charge_loop_req: DCChargeLoopReq,
+        selected_energy_service: SelectedEnergyService,
+        control_mode: ControlMode,
+    ) -> None:
+        if selected_energy_service.service == ServiceV20.DC:
+            if control_mode == ControlMode.SCHEDULED:
+                self.comm_session.evse_controller.ev_data_context.ev_session_context.update(  # noqa
+                    dc_charge_loop_req.scheduled_params
+                )
+            elif control_mode == ControlMode.DYNAMIC:
+                self.comm_session.evse_controller.ev_data_context.ev_session_context.update(  # noqa
+                    dc_charge_loop_req.dynamic_params
+                )
+        elif selected_energy_service.service == ServiceV20.DC_BPT:
+            if control_mode == ControlMode.SCHEDULED:
```

Comment:
it may be a bit of a hacky/obfuscated way to do this, but what if we do:
```python
# this gets the scheduled or dynamic function attribute
bpt_attributes = getattr(dc_charge_loop_req, 'bpt_' + control_mode.name.lower())
self.comm_session.evse_controller.ev_data_context.ev_session_context.update( 
                    bpt_attributes
```

It saves some lines of code, but maybe turns the code less readable and prone to error if in the future we change the attributes names. What do you think?

Reply:
Yea - right now even though there are a few more lines, what is expected is evident. The above change would break if we change the field names like you said...